### PR TITLE
Style refactor

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
When creating new dialogs and widgets, I have often noticed the need to add widget-specific styles (e.g. padding around a `QPushButton`, background on a `QLabel` or something like that).  This makes it difficult to standardize the look & feel of widgets within napari, and will also make it hard for others to extend the napari GUI (e.g. with `magicgui`) without needing to tweak their own stylesheets.  This PR refactors our stylesheets (and other style related things) with the aim of standardizing styles without needing widget-specific styles (for instance, a "spinbox" in napari, should look like all the other ones).  I'm mostly done, and will remove [WIP] when it's ready for review.

**This PR:**
- adds a `SampleWidget` in `_qt/theme_sample.py` that contains most other widget types, which could be used to show a theme preview or test out a new theme.
- breaks up `stylesheet.qss` into multiple cascading sheets for easier organization.
    - base.qss refers only to built-in Qt types, and establishes the base themed style.
    - other stylesheets in there are used to customize napari-specific objects.
- removes as many duplicate/conflicting styles as possible.
- adds two new `darken`/`lighten` "directives" that can be used in stylesheets.  For instance the following will create a color that is 15 percent darker than `{{ background }}`.
```css
QLineEdit {
  background-color: {{ darken(background, 15) }};
}
```
- overhauls `build_icons.py` to include _all_ required steps for adding an icon.  Now adding an icon simply requires putting the svg file in `/icons/svgs` and running `python -m napari.resources.build_icons`.  `CONTRIBUTING.md` has been updated with new instructions.

before after comparison of the test Widget:
before_after

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?
very little functional code has been touched.

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by tlambert03** [comment]
one additional thing that might be worth considering in this PR is whether we want to move to relative `rem` or `em` units, as briefly discussed in https://github.com/napari/napari/pull/820#issuecomment-574440426.  (Note: I tried using `pt` and I don't think it works, but rem/em might work).

File: napari/_qt/theme_sample.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
+"""SampleWidget that contains many types of QWidgets.
+
+This file and SampleWidget is useful for testing out themes from the command
+line or for generating screenshots of a sample widget to demonstrate a theme.
+
+Examples
+--------
+To use from the command line:
+
+$ python -m napari._qt.theme_sample
```
Comment (sofroniewn) [inline napari/_qt/theme_sample.py:10]: Might be nice to put this in our contributing guidelines in a section on themes? I just ran it, it's pretty cool!!!!

File: docs/developers/CONTRIBUTING.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ If you wish to tell the linter to ignore a specific line use the `# noqa` commen
 
 ## Building the icons
 
-Our icon build process starts with a base set of `svg` formatted icons stored in `napari/resources/icons/svg`. If you want to add a new icon to the app, make the icon in whatever program you like, add it to that folder, and make sure it is listed in both `napari/resources/build_icons.py` and `napari/resources/res.qrc`.
+Our icon build process starts with a base set of `svg` formatted icons stored in `napari/resources/icons/svg`. If you want to add a new icon to the app, make the icon in whatever program you like and add it to that folder.
 
 Then run the following command
```
Comment (sofroniewn) [inline docs/developers/CONTRIBUTING.md:52]: So nice just to have one command now!!! Makes dev much easier!!!!

File: napari/_qt/theme_sample.py (hunk 1 above)
Comment (sofroniewn) [reply napari/_qt/theme_sample.py:10]: also can you make it so if you pass a string arg corresponding to one of our known themes only that one is displayed, i.e.

This only shows the dark theme
```
python -m napari._qt.theme_sample dark
```

This only shows the light theme
```
python -m napari._qt.theme_sample light
```

and this shows all
```
python -m napari._qt.theme_sample
```

**Review by sofroniewn — APPROVED** [review]
This looks great @tlambert03!! I love that only one command is now needed to rebuild the icons, and I love the theme_samples. I left only one minor suggestion there.

**Comment by tlambert03** [comment]
ok, I fixed some of the last remaining things I wanted to and removed WIP.  Probably easiest to check this one out and play with it when reviewing.

File: napari/_qt/theme_sample.py (hunk 1 above)
Comment (tlambert03) [reply napari/_qt/theme_sample.py:10]: totally agree. I actually had that at one point, disabled it for some other reason, then never brought it back.  will add

Comment (tlambert03) [reply napari/_qt/theme_sample.py:10]: did both

**Comment by sofroniewn** [comment]
Love this, thanks for updating the docs. Curious how the high DPI stuff is looking?

**Comment by tlambert03** [comment]
well, it looks ok on my linux box with a high DPI monitor, but only if I export QT_ENABLE_HIGHDPI_SCALING.  I can't remember if one of our hopes was that simply using `pt` instead of `px` would make that unnecessary?  But at least it looks good with it enabled.
The one issue I'm still having is that the total slice label (bottom right) is _slightly_ off centered on windows and linux high DPI.  but I can't fix it while keeping all the other platforms aligned 😜

**Comment by sofroniewn** [comment]
> I can't remember if one of our hopes was that simply using pt instead of px would make that unnecessary?

Yes that was a hope as setting `QT_ENABLE_HIGHDPI_SCALING` had to be done before application launch, though there was a recent comment that said that changed in a new Qt release.

We'll let @jni take a look before merge, and you can wait till you're happy tweaking, but this is looking great :-)

**Comment by tlambert03** [comment]
ok, I wrote up guidelines for writing tests with proper cleanup and added them to contributing.md [here](https://github.com/napari/napari/blob/97c584f21f426c03477454254c9dd5da38df5694/docs/developers/CONTRIBUTING.md#writing-tests)

**Comment by sofroniewn** [comment]
> ok, I wrote up guidelines for writing tests with proper cleanup and added them to contributing.md here

Ah @tlambert03 this comment above was meant for another PR - hence https://github.com/napari/napari/pull/1014#issuecomment-594242206

> (thought I posted this yesterday but I guess not!)

I'll let this sit one more day and then merge :-)

**Comment by tlambert03** [comment]
lol!  there it is!  i was wondering where that went 😜

**Review by jni — APPROVED** [review]
Glorious! I certainly can't say that I've seen every line of code, but I love the new test widget and the new look! =)

**Comment by tlambert03** [comment]
> Glorious! I certainly can't say that I've seen every line of code, but I love the new test widget and the new look! =)

you mean you didn't look through all of the byte code changes in `qt.py` ??

**Pull request merged**